### PR TITLE
Add collect_hub function

### PR DESCRIPTION
**Pull request by annakrystalli**
This PR adds a `collect_hub()` function which wraps `dplyr::collect()` but also converts the output to a `model_out_tbl` class object by default where possible. The function also accepts additional arguments that can be passed to `as_model_out_tbl()`.

Th PR resolves:
- [x] #17
- [x] #15
- [x] #4 
- [x] #13 

I've also modified the R CMD Check workflow to run nightly so that we can pick up any issues arising from upgrades in dependencies promptly. Let's test it out and I can roll it out to all our mature packages when ready.

### CODE REVIEW

**Comment by annakrystalli** [comment]
The macos latest build is failing because of #15 . Not sure what to do about it. I could fix it by installing from the Apache R Universe version in the workflow. That's not where most users would do though so, until we swapped it back to CRAN, we 
a. wouldn't get errors  representative of median user experience
b. I won't know when the issue is fixed unless I explictly track when a macOS arrow fix is pushed to CRAN.

Happy to hear people's thoughts.

**Comment by bsweger** [comment]
Leaving this here for other reviewers who might want to fetch this feature branch and give it a spin locally. Where `hubverse-infrastructure-test` is an S3 bucket with valid `hub-config` contents and an empty `model-output` directory:

**dplyr::collect()** 😢 

```
> load_all()
ℹ Loading hubData
> hub_bad_path <- s3_bucket('hubverse-infrastructure-test')
> hubData::connect_hub(hub_path = hub_bad_path) %>% dplyr::collect()
Error in UseMethod("collect") : 
  no applicable method for 'collect' applied to an object of class "c('hub_connection', 'list')"
In addition: Warning message:
In hubData::connect_hub(hub_path = hub_bad_path) :
  No files of file formats "csv", "parquet", and "arrow" found in model output directory.
```

**hub_collect()** 😀 
```
> tbl <- connect_hub(hub_bad_path) %>% collect_hub()
Warning messages:
1: In connect_hub(hub_bad_path) :
  No files of file formats "csv", "parquet", and "arrow" found in model output directory.
2: Hub is empty. No data to collect. Returning `NULL` 
```

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -5,6 +5,8 @@ on:
     branches: [main, master]
   pull_request:
     branches: [main, master]
+  schedule:
```
Comment (bsweger) [inline .github/workflows/R-CMD-check.yaml:8]: Makes sense! Would love to get to a place where these kinds of small operational changes can be in separate PRs so we can get 'em merged in without waiting for review of new features.

File: vignettes/articles/connect_hub.Rmd [hunk 2]
```diff
@@ -54,12 +54,26 @@ hub_con
 
 To access data from a hub connection you can use dplyr verbs and construct querying pipelines.
 
+You can use `dplyr`'s `collect()` function:
```
Comment (bsweger) [inline vignettes/articles/connect_hub.Rmd:57]: Now that we have `collect_hub()`, is there any reason someone would want to use dplyr collect()?

As someone with less R proficiency than many folks on the team, I'm left wondering what to do when presented with multiple options like this. Is it work recommending a default?

**Review by bsweger — APPROVED** [review]
Thanks, @annakrystalli. Glad we're trying to handle the hard bits on behalf of users!

One or two inline notes, but nothing that would prevent rolling out this improvement.

File: vignettes/articles/connect_hub.Rmd (hunk 2 above)
Comment (annakrystalli) [reply vignettes/articles/connect_hub.Rmd]: `collect_hub` is mainly a wrapper around `dplyr::collect()` (a very well known tidyverse function) with some extras. It depends on what they are doing with their data next but there is no reason they **must** use `collect_hub()`, it just conveniently outputs a `model_out_tbl` which many downstream hubverse package functions expect.

I've refactored the article a bit to bring more attention to the benefits of `collect_hub` and also used it in the `connect_hub` examples but ultimately `collect` will work just as well. It just might need an extra step to coerce data to `model_out_tbl` if used in downstream `hubverse` functionality
